src/d_main.c

```diff
@@ -1741,6 +1741,49 @@ void D_SetBloodColor(void)
   }
 }
 
+void D_SetTranslucency(int index)
+{
+  int i;
+
+  const int predefined_translucency[] = {
+    // MBF
+    MT_FIRE,      MT_SMOKE,     MT_FATSHOT,  MT_BRUISERSHOT,
+    MT_SPAWNFIRE, MT_TROOPSHOT, MT_HEADSHOT, MT_PLASMA,
+    MT_BFG,       MT_ARACHPLAZ, MT_PUFF,     MT_TFOG,
+    MT_IFOG,      MT_MISC12,    MT_INV,      MT_INS,
+    MT_MEGA,
+    // [Woof!]
+    MT_PLASMA1,   MT_PLASMA2
+  };
+
+  static boolean deh_set_translucency[arrlen(predefined_translucency)] = {false};
```

Comment:
We could also use an array of `struct {int; boolean;}`, but that's up to you.

Reply:
But then the initialization will be more verbose: `{ {MT_FIRE, false}, ... }`

Reply:
Yes, right.